tasks/misc/review-incentive.ts

```diff
@@ -0,0 +1,183 @@
+    if (!oracleAddr || oracleAddr == ZERO_ADDRESS ) {
+      console.log(chalk.red(`'${network}.${reserve}[${i}]': oracle wasn't found in ChainlinkAggregator or is not valid`));
+      exit(1);
+    }
+
+    if (cfg.transferStrategy != TransferStrategy.PotRewardsStrategy) {
+      console.log(chalk.red(`'${network}.${reserve}[${i}]': invalid transfer strategy. Only PotRewardsStrategy is supported`));
+      exit(1);
+    }
+
+    if (!cfg.reward || cfg.reward == ZERO_ADDRESS) {
+      console.log(chalk.red(`'${network}.${reserve}[${i}]': invalid reward address`));
+      exit(1);
+    }
+
+    if ( cfg.emissionAdmin.toLowerCase() != admin.toLowerCase()) {
+      console.log(chalk.red(`'${network}.${reserve}[${i}]': emission admin is not set to poolAdmin`));
+      exit(1);
+    }
+
+    const onChainInc = onChainIncentives.find((el) => el.underlyingAsset == reserveAddr);
+    if (!onChainInc) {
+      //NOTE: onChainInc should always exists even if incentives were never deployed
+      console.log(chalk.red(`'${network}.${reserve}[${i}]': unexpected error. No incentives data found onchain.`));
+      exit(1);
+    }
+
+    const {
+      aTokenAddress,
+      stableDebtTokenAddress,
+      variableDebtTokenAddress,
+    } = await dataProvider.getReserveTokensAddresses(reserveAddr);
+
+    let activeInc;
+    let asset;
+    switch (cfg.assetType) {
+      case AssetType.AToken:
+        activeInc = onChainInc.aIncentiveData.rewardsTokenInformation.find(el => el.rewardTokenAddress == cfg.reward );
+        asset = aTokenAddress;
+        break;
+      case AssetType.VariableDebtToken:
+        activeInc = onChainInc.vIncentiveData.rewardsTokenInformation.find(el => el.rewardTokenAddress == cfg.reward );
+        asset = variableDebtTokenAddress;
+        break;
+      case AssetType.StableDebtToken:
+        activeInc = onChainInc.sIncentiveData.rewardsTokenInformation.find(el => el.rewardTokenAddress == cfg.reward );
+        asset = stableDebtTokenAddress;
+        break;
+      default:
+        console.log(chalk.red(`'${network}.${reserve}[${i}]': unknown assetType option: ${cfg.assetType}`));
+        exit(1);
+    }
+
+
+    const now = await getBlockTimestamp();
+
+    if (activeInc?.emissionEndTimestamp.gt(BigNumber.from(now)) && !update) {
+      console.log(chalk.yellow(`'${network}.${reserve}[${i}]': liquidity mining is active. To update it use --update`));
+      continue;
+    }
+
+    //start or update incentives
+    assetsConf.push({
+      emissionPerSecond: cfg.emissionPerSecond,
+      distributionEnd: now + cfg.duration,
+      asset: asset,
+      reward: cfg.reward,
+      transferStrategy: transferStrat.address,
+      rewardOracle: oracleAddr,
+      totalSupply: "0",
+    });
+  }
+
+  if (assetsConf.length != 0) {
+    const tx = await em.populateTransaction.configureAssets(assetsConf, { gasLimit: 1000000 });
+    addTransaction(tx);
+  }
+
+  if (batch) {
+    return
+  }
+
+  const txs = getBatch();
```

Comment:
i guess the rest of this can be removed after the batch check

the batch flag is usually used to call `addTransaction` instead of directly sending a transaction from the deployer account and waiting for the result. Proposal generation shouldn't be the responsibility of this task.